demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/products/ProductDetails.kt

```diff
@@ -71,3 +77,80 @@ fun ProductDetails(product:Product){
         }
     }
 }
+
+@Composable
+fun QuantityChooser() {
+    var quantity by remember { mutableIntStateOf(1) }
+    var expanded by remember { mutableStateOf(false) }
+
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier.padding(vertical = 8.dp)
+    ) {
+        Text(
+            text = "Quantity",
+            fontFamily = gothamFont,
+            fontSize = 18.sp,
+            modifier = Modifier.padding(start = 16.dp)
+        )
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp)
+        ) {
+            OutlinedButton(
+                onClick = { expanded = true },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(48.dp)
+            ) {
+                Row(
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+                    Text(
+                        text = quantity.toString(),
+                        fontSize = 18.sp,
+                        modifier = Modifier.align(Alignment.CenterVertically)
+                    )
+                    Icon(
+                        imageVector = Icons.Filled.ArrowDropDown,
+                        contentDescription = "Dropdown Icon",
+                        modifier = Modifier.align(Alignment.CenterVertically)
+                    )
+                }
+            }
+            DropdownMenu(
+                expanded = expanded,
+                onDismissRequest = { expanded = false }
+            ) {
+                for (i in 1..10) {
+                    DropdownMenuItem(
+                        text = {
+                            Row(
```

Comment:
nit: I think we should (whenever possible) try to split big widgets into multiple composable functions to avoid "indentation hell".